infrastructure/lambda/template.yaml

```diff
@@ -632,6 +636,52 @@ Resources:
 #                                                                  #
 ####################################################################
 
+  DLDrivingPermitCheckLambdaConcurrencyThresholdReached:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      AlarmDescription: !Sub Driving Permit CRI ${Environment} - DrivingPermitCheck Lambda concurrency threshold reached
+      ActionsEnabled: true
+      AlarmActions:
+        - !Ref AlarmTopicDL
+      OKActions:
+        - !Ref AlarmTopicDL
+      InsufficientDataActions: []
+      MetricName: ConcurrentExecutions
+      Namespace: AWS/Lambda
+      Statistic: Sum
+      Dimensions:
+        - Name: FunctionName
+          Value: !Ref DrivingPermitCheckingFunction
+      Period: 60
+      EvaluationPeriods: 15
+      DatapointsToAlarm: 15
+      Threshold: 2
+      ComparisonOperator: GreaterThanOrEqualToThreshold
+      TreatMissingData: notBreaching
+
+  DLIssueCredentialLambdaConcurrencyThresholdReached:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      AlarmDescription: !Sub Driving Permit CRI ${Environment} - IssueCredential Lambda concurrency threshold reached
+      ActionsEnabled: true
+      AlarmActions:
+        - !Ref AlarmTopicDL
+      OKActions:
+        - !Ref AlarmTopicDL
+      InsufficientDataActions: []
+      MetricName: ConcurrentExecutions
+      Namespace: AWS/Lambda
+      Statistic: Sum
+      Dimensions:
+        - Name: FunctionName
+          Value: !Ref IssueCredentialFunction
+      Period: 60
```

Comment:
This is going to be running checks every 4 seconds can we not mimic the other alarms and say
period:300
EvaluationPeriods: 3
EvaluationPeriods: 3
Threshold: 3 ( Going up by one I still think is fine so we can take contingency)
Then sustained load over a 5 minute window alerts us.

If were worried about peaks we could do one with a smaller window and a higher threshold

Reply:
Tweaked this to use the average of concurrentExecutions.
ConcurrentExecutions >= 2 for 12 datapoints within 15 minutes (sampled every minute) will alarm at the 80% (12/15) sustained traffic increase and ignore temporary large bursts.

Reply:
Threshold amended to ConcurrentExecutions >= 3